### DISCUSSION TITLE
uhppoted-httpd

### DISCUSSION THREAD

**Discussion by liviom** [uhppoted/uhppoted#13] [Q&A] [answered]
It's always me :-)

I saw the website activated running uhppoted-httpd  
the informations are loaded from json db in /var/uhppoted....
but db (events.json) it's not automatically update from the board?

How can I bind events.json with the board to automatically download the events?

**Comment by uhppoted** [comment]
Ah! You figured out how to delete records by e.g. clearing both name and card number :-) 

(the logic is that on an access control system you seldom want to delete both a person and a card - mostly you want to keep the card active but change who it is assigned to, or alternatively change the card number for somebody who has lost their card)

re. events. You don't need to do anything special - they are retrieved automatically in small batches at regular intervals (30 seconds by default).

**Comment by liviom** [comment]
Maybe my configuration doesn't works as expected because eventi are frozen and new events are not downloaded/displayed

How can I check?

Sent from Nine
________________________________
Da: uhppoted ***@***.***>
Inviato: Venerdì 8 Luglio 2022 18:53
A: uhppoted/uhppoted
Cc: liviom; Author
Oggetto: Re: [uhppoted/uhppoted] uhppoted-httpd (Discussion #13)

Ah! You figured out how to delete records by e.g. clearing both name and card number :-)
(the logic is that on an access control system you seldom want to delete both a person and a card - mostly you want to keep the card active but change who it is assigned to, or alternatively change the card number for somebody who has lost their card)
re. events. You don't need to do anything special - they are retrieved automatically in small batches at regular intervals (30 seconds by default).

**Comment by uhppoted** [comment]
Out of the box _uhppoted-httpd_ is reasonably robust configuration-wise unless you've got something weird set up, so it's probably not that.

Can you take a look at your logs? If you're running it console mode (`./bin/uhppoted-httpd --debug --console`) then the logger will be writing to the terminal, otherwise it will be writing to _/var/uhppoted/logs/uhppoted-httpd.log_ (or the equivalent on your platform).

You should see log events that look like this:
```
2022/07/08 12:22:38 INFO  201020304: retrieving LAN controller events ([{1006 4294967295} {1 1000}])
2022/07/08 12:22:38 INFO  303986753: retrieving LAN controller events ([{1 4294967295}])
2022/07/08 12:22:38 INFO  405419896: retrieving LAN controller events ([{11 4294967295}])
```

If all else fails:
1. Stop _uhppoted-httpd_
2. Delete _events.json_ 
3. And then run _uhppoted-httpd_ again

**Reply by liviom** [reply]
this is the log

`2022/07/09 00:59:05 WARN  INTERNAL SERVER ERROR: Error getting door 1 control for 423133701 (Error creating UDP socket (listen udp 192.168.0.101:0: bind: cannot assign requested address))
2022/07/09 00:59:05 DEBUG get-door-control request  {DeviceID:423133701 Door:2}
2022/07/09 00:59:05 WARN  INTERNAL SERVER ERROR: Error getting door 2 control for 423133701 (Error creating UDP socket (listen udp 192.168.0.101:0: bind: cannot assign requested address))
2022/07/09 00:59:05 DEBUG get-door-control request  {DeviceID:423133701 Door:3}
2022/07/09 00:59:05 WARN  INTERNAL SERVER ERROR: Error getting door 3 control for 423133701 (Error creating UDP socket (listen udp 192.168.0.101:0: bind: cannot assign requested address))
2022/07/09 00:59:05 DEBUG get-door-control request  {DeviceID:423133701 Door:4}
2022/07/09 00:59:05 WARN  INTERNAL SERVER ERROR: Error getting door 4 control for 423133701 (Error creating UDP socket (listen udp 192.168.0.101:0: bind: cannot assign requested address))
2022/07/09 00:59:05 INFO  423133701: retrieving LAN controller events ([{1 4294967295}])
2022/07/09 00:59:05 WARN  Error creating UDP socket (listen udp 192.168.0.101:0: bind: cannot assign requested address)
2022/07/09 00:59:05 INFO  Sweeping all items invalidated before 2022-07-08 18:59:05
2022/07/09 00:59:05 DEBUG Comparing ACL
2022/07/09 00:59:05 WARN  Error creating UDP socket (listen udp 192.168.0.101:0: bind: cannot assign requested address)
2022/07/09 00:59:05 INFO  ACL 423133701  unchanged:0   updated:0   added:0   deleted:2
2022/07/09 00:59:05 INFO  ACL compare    unchanged:0   updated:0   added:0   deleted:2
`

it seems that can't assign udp socket...
maybe I set something wrong?

I run it as root user with sudo command...

**Comment by uhppoted** [comment]
If I remember your configuration correctly:
- your Ubuntu server IP address is 192.168.0.90
- your Windows server IP address is 192.168.0.70
- your controller IP is 192.168.0.101

so it looks like you've assigned the controller address to the LAN bind address which is why it cannot bind. On the _SYSTEM_ page:

1. Set the LAN _bind_ address to either 192.168.0.70 or 192.168.0.90 (depending on whether uhppoted-httpd is running on the Ubuntu box or the Windows box
2. Set the LAN _broadcast_ address to either 192.168.0.255 or 192.168.0.255 (this may not be correct - check your network configuration for the actual broadcast address)

Although, TBH, you could probably just go with the generic values:
- bind: 0.0.0.0
- broadcast: 255.255.255.255

<img width="1430" alt="Screenshot 2022-07-08 at 17 40 45" src="https://user-images.githubusercontent.com/61560842/178085095-bf1c82ed-717b-4dc9-a02c-c8fa2349f413.png">

**Reply by liviom** [reply]
Damn, It's me the only fault :-)... I was changing the bind address in uhppoted.conf file....changing in interface.json file works perfectly....


the last questions ( I hope)

How to change columns width and fonts? 

![uhppoted5](https://user-images.githubusercontent.com/13430820/178091226-8f99a72d-e6d8-4be6-869b-f651994b0de8.png)


maybe fonts in css files in /etc/uhppoted/httpd/html but i didn't find anything about column width at first view....



why events are download until 01/07 and no other events are downloaded until today?

![uhppoted7](https://user-images.githubusercontent.com/13430820/178092284-81b85316-0a44-47a2-ba2a-6550d18e4d44.png)

![uhppoted6](https://user-images.githubusercontent.com/13430820/178092339-4dbe649f-8e22-4729-bf7a-a9cc5ee3cb25.png)

and what about this???

_
ERRO[0094] Max cycle reached                             lib=grule-rule-engine package=engine
2022/07/09 07:20:14 WARN  the GruleEngine successfully selected rule candidate for execution after 5000 cycles, this could possibly caused by rule entry(s) that keep added into execution pool but when executed it does not change any data in context. Please evaluate your rule entries "When" and "Then" scope. You can adjust the maximum cycle using GruleEngine.MaxCycle variable
_

thanks again

**Reply by liviom** [reply]
maybe there is a bug in  downloading events in uhppoted-httpd? 
I noticed that the Not Found WARN is related to reboot event of the board, this warn seems doesn't allow to unload more events.

![uhppoted8](https://user-images.githubusercontent.com/13430820/178093856-f3cc67fc-cd07-4963-881b-560adeb7278b.png)

I added the record manually in events.json and finally the events page is updated....

**Comment by uhppoted** [comment, marked as answer]
re. fonts/column widths. There's a write-up on modifying the HTML [here](https://github.com/uhppoted/uhppoted-httpd/blob/master/documentation/HTML.md)

re. reboot event. Hmm, interesting. I'll have to look into it.

re. grules error. That typically happens when one of the rules is invalid. Have you modified any of the grules file?

**Comment by liviom** [comment]
Yes I try to modify a grules....

The sintax is similar to openhab rules, isn't it?


Sent from Nine
________________________________
Da: uhppoted ***@***.***>
Inviato: Sabato 9 Luglio 2022 18:07
A: uhppoted/uhppoted
Cc: liviom; Author
Oggetto: Re: [uhppoted/uhppoted] uhppoted-httpd (Discussion #13)

re. fonts/column widths. There's a write-up on modifying the HTML here
re. reboot event. Hmm, interesting. I'll have to look into it.
re. grules error. That typically happens when one of the rules is invalid. Have you modified any of the grules file?

**Comment by uhppoted** [comment]
> The sintax is similar to openhab rules, isn't it?

Not even close! Hypertech grule-engine - the documentation is [here](https://github.com/uhppoted/uhppoted-httpd/blob/master/documentation/grules.md)

**Comment by uhppoted** [comment]
Hmmm, it appears there is a bug in the UHPPOTE controller firmware around the power-on events (cf. [Retrieve events over multiple missing events boundary](https://github.com/uhppoted/uhppoted-httpd/issues/8)) where it doesn't return a valid event for power-on. 

It shouldn't affect most people - it showed up on your system because you had more than 5 consecutive power on events. I'll have to rethink that logic a little to workaround the problem.

**Comment by liviom** [comment]
dear

I reinstall from zero to another clean vm and I can not login as default user with user admin and  uhppoted password.
The message is  "Invalid user ID or password" from html login page and from console :



2022/07/12 21:22:31 DEBUG HEAD /authenticate
2022/07/12 21:22:31 DEBUG POST /authenticate
2022/07/12 21:22:31 WARN  Invalid login credentials


Where I'm wrong?

when I launched the uhppoted-httpd daemonize I answer also yes when he asked me to create the default admin user

**Comment by YBonline** [comment]
I just ran into this, it should have gave you the password when it asked if you wanted to create the default admin user, if you didn't save that password, you're going to want to delete /var/uhppoted/httpd/system/users.json and run the install again. You also want to make sure there is nothing the users section of etc/uhpptoed/auth.json (you could also delete this file and let it regenerate it when you reinstall). uhppoted isn't actually the default password except in the example config file

**Comment by uhppoted** [comment]
Thanks @YBonline :-). I've created an issue to make the admin user ID and password a bit more obvious because it's clearly going to be an ongoing problem:

- [Display the generated admin user/password as the last step of daemonize](https://github.com/uhppoted/uhppoted-httpd/issues/9)

@liviom, the issue @YBonline is referring to has more detail:

- [default auth doesn't work](https://github.com/uhppoted/uhppoted-httpd/issues/6)

**Comment by uhppoted** [comment]
Hi,

Not sure if you're still using _uhppoted-httpd_, but please see the security notice in the [README](https://github.com/uhppoted/uhppoted-httpd?tab=readme-ov-file#uhppoted-httpd).

Most people are using it with a single user but if you have allowed access to multiple users you should consider upgrading to the latest _Docker_ image on _ghcr.io_ which has been updated with the fix.

Apologies if it causes any inconvenience :-(.
